src/Common/src/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -1395,9 +803,42 @@ protected virtual void EmitMarshalElementNativeToManaged()
             LoadManagedValue(codeStream);
         }
 
+        protected virtual void EmitMarshalFieldManagedToNative()
+        {
+            ILEmitter emitter = _ilCodeStreams.Emitter;
+            ILCodeStream marshallingCodeStream = _ilCodeStreams.MarshallingCodeStream;
+
+            SetupArgumentsForFieldMarshalling();
+
+            StoreManagedValue(marshallingCodeStream);
```

Comment:
Some comments here would be nice. You are expecting the field value already locked on stack.